python/lsst/ip/diffim/detectAndMeasure.py

```diff
@@ -376,6 +383,33 @@ def processResults(self, science, matchedTemplate, difference, sources, table,
 
         return measurementResults
 
+    def removeBadSources(self, diaSources):
+        """Remove bad diaSources from the catalog.
+
+        Parameters
+        ----------
+        diaSources : `lsst.afw.table.SourceCatalog`
+            The catalog of detected sources.
+
+        Returns
+        -------
+        diaSources : `lsst.afw.table.SourceCatalog`
+            The updated catalog of detected sources, with any source that has a
+            flag in ``config.badSourceFlags`` set removed.
+        """
+        flags = np.ones(len(diaSources), dtype=bool)
+        for flag in self.config.badSourceFlags:
+            try:
+                flags &= ~diaSources[flag]
+            except Exception as e:
+                self.log.warning("Could not apply source flag: %s", e)
```

Comment:
I don't like this try:catch; if you want to not raise on missing flags, you should only catch `KeyError`. More broadly, I think a missing flag should just raise: if someone configures their pipeline to expect a certain flag, it should be an error if it doesn't appear in the schema, otherwise it's hiding unintended behavior, like a typo in the flags list.

Reply:
I understand your objection to a try:catch here, and can certainly change this to a `KeyError`, but I would strongly prefer keeping it. My reasoning is that this should ideally not be finding and removing any bad sources, so I would rather it shout at people with a warning than have it break if the configured flag is missing.

Reply:
I very strongly disagree. An error that occurs on every single quantum (which is what will happen if the config doesn't match the schema) should be an actual exception raised up the chain, not a warning that will get lost. If it's an error, processing will stop early and we'll know to fix the misconfiguration. If it's a warning, we could process a whole data release without knowing that we meant to filter on something but didn't.

I'm not sure what you mean by "should ideally not be finding and removing any bad sources"? We know there will be "bad sources" because difference imaging will produce totally spurious detections due to interpolation. Unless we remove those sources much earlier (which is hard to do, given the design of the plugin system), we know they'll show up here and have to be removed.

Reply:
There's actually a special exception type for configuration/consistency errors that can only be caught during after execution (whenever possible we want to catch them in config validation or task construction, of course): [`lsst.pipe.base.InvalidQuantumError`](https://github.com/lsst/pipe_base/blob/9d3cecc6b43000bf6755f3880c79acf7bc896de6/python/lsst/pipe/base/_status.py#L77C7-L77C26).  I don't know the context well enough to say for certain whether it's the right call, but it seems potentially relevant.

Reply:
After a side discussion with @parejkoj , lsst.pipe.base.InvalidQuantumError in the task `__init__` looks like the right solution here.